stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2786,6 +2786,25 @@ extension Dictionary : Equatable where Value : Equatable {
   }
 }
 
+extension Dictionary : Hashable where Value : Hashable {
+  /// The hash value for the dictionary.
+  ///
+  /// Two dictionaries that are equal will always have equal hash values.
+  ///
+  /// Hash values are not guaranteed to be equal across different executions of
+  /// your program. Do not save hash values to use during a future execution.
+  @_inlineable // FIXME(sil-serialize-all)
+  public var hashValue: Int {
+    // FIXME(ABI)#177: <rdar://problem/18915294> Issue applies to Dictionary too
+    var result: Int = _mixInt(0)
+    for (k, v) in self {
+      let combined = _combineHashValues(k.hashValue, v.hashValue)
+      result ^= _mixInt(combined)
```

Comment:
I think `_combineHashValues()` rather than `^=` should be used.

Reply:
A previous implementation did this, but it was commented that this makes the hash depend on the traversal order. 

So that two equal dicts that happen to be traversed in different order would produce different hashes.